jetty-websocket/websocket-core-server/src/main/java/org/eclipse/jetty/websocket/core/server/internal/AbstractHandshaker.java

```diff
@@ -167,6 +167,7 @@ public boolean upgradeRequest(WebSocketNegotiator negotiator, HttpServletRequest
         // Save state from request/response and remove reference to the base request/response.
         upgradeRequest.upgrade();
         upgradeResponse.upgrade();
+        negotiation.setBaseRequest(null);
```

Comment:
The previous 2 lines that call `upgrade()` actually null-out the references to the request and response, respectively.
I would call the method on `WebSocketNegotiation` as `upgrade()` as well, just in case in future we want to null-out some other field, or copy something from the `Request` object to save it for later use, like the other `upgrade()` methods do.